samples/server/petstore/nodejs/controllers/Pet.js

```diff
@@ -1,37 +1,97 @@
 'use strict';
 
-var url = require('url');
-
-var Pet = require('./PetService');
+require('../utils/writer.js');
+var Pet = require('../service/PetService');
 
 module.exports.addPet = function addPet (req, res, next) {
-  Pet.addPet(req.swagger.params, res, next);
+    var body = req.swagger.params['body'].value;
+    Pet.addPet(body)
+        .then(function (response) {
+            writeJson(res, response);
+        })
+        .catch(function (response) {
+            writeJson(res, response);
+        });
 };
 
 module.exports.deletePet = function deletePet (req, res, next) {
-  Pet.deletePet(req.swagger.params, res, next);
+    var petId = req.swagger.params['petId'].value;
+    var api_key = req.swagger.params['api_key'].value;
+    Pet.deletePet(petId,api_key)
+        .then(function (response) {
+            writeJson(res, response);
+        })
+        .catch(function (response) {
+            writeJson(res, response);
+        });
 };
 
 module.exports.findPetsByStatus = function findPetsByStatus (req, res, next) {
-  Pet.findPetsByStatus(req.swagger.params, res, next);
+    var status = req.swagger.params['status'].value;
+    Pet.findPetsByStatus(status)
+        .then(function (response) {
+            writeJson(res, response);
+        })
+        .catch(function (response) {
+            writeJson(res, response);
+        });
 };
 
 module.exports.findPetsByTags = function findPetsByTags (req, res, next) {
-  Pet.findPetsByTags(req.swagger.params, res, next);
+    var tags = req.swagger.params['tags'].value;
+    Pet.findPetsByTags(tags)
+        .then(function (response) {
+            writeJson(res, response);
+        })
+        .catch(function (response) {
+            writeJson(res, response);
+        });
 };
 
 module.exports.getPetById = function getPetById (req, res, next) {
-  Pet.getPetById(req.swagger.params, res, next);
+    var petId = req.swagger.params['petId'].value;
+    Pet.getPetById(petId)
+        .then(function (response) {
+            writeJson(res, response);
+        })
+        .catch(function (response) {
+            writeJson(res, response);
+        });
 };
 
 module.exports.updatePet = function updatePet (req, res, next) {
-  Pet.updatePet(req.swagger.params, res, next);
+    var body = req.swagger.params['body'].value;
+    Pet.updatePet(body)
+        .then(function (response) {
+            writeJson(res, response);
+        })
+        .catch(function (response) {
+            writeJson(res, response);
+        });
 };
 
 module.exports.updatePetWithForm = function updatePetWithForm (req, res, next) {
-  Pet.updatePetWithForm(req.swagger.params, res, next);
+    var petId = req.swagger.params['petId'].value;
+    var name = req.swagger.params['name'].value;
+    var status = req.swagger.params['status'].value;
```

Comment:
note, all input values are extracted in generated code to make it easier to implement the service class.